### PR TITLE
add near-term China historical data of gas power plants

### PR DESCRIPTION
**Pull request by cchrisgong**
This PR is to replace the hard coded input data in this PR: https://github.com/remindmodel/remind/pull/1194

### CODE REVIEW

**Comment by cchrisgong** [comment]
Hi Lavinia, I get the data now in mrremind. If looks okay I will update version number, etc. Danke~

File: R/calcCapacity.R [hunk 1]
```diff
@@ -72,12 +72,19 @@ calcCapacity <- function(subtype) {
     # renaming technologies to REMIND naming convention
     WEOcap <- rename_dimnames(WEOcap, dim = 3, query = mapping, from = "WEO_techs", to="REMIND_techs")
     WEOcap <- WEOcap * 1E-03 # converting GW to TW
-
+    
+#*** near term China gas data - see below
```
Comment (robertpietzcker) [inline R/calcCapacity.R:76]: maybe add a comment: "creating empty lines to be filled by hand below"?

File: R/calcCapacity.R [hunk 2]
```diff
@@ -156,7 +185,23 @@ calcCapacity <- function(subtype) {
     output[getRegions(WEOcap),getYears(WEOcap),getNames(WEOcap)] <- WEOcap[getRegions(WEOcap),
                                                                            getYears(WEOcap),
                                                                            getNames(WEOcap)]
-
+    
+    #    ***CG: fix CHA gas power capacities: 97 GW by September 2020 (Oxford Institute for Energy Studies:
```
Comment (robertpietzcker) [inline R/calcCapacity.R:189]: I don't really understand these scripts - just wondering why you need to input the data twice?

**Review by robertpietzcker — APPROVED** [review]
I assume you checked the output and it contains the right values :-)

File: R/calcCapacity.R (hunk 1 above)
Comment (cchrisgong) [reply R/calcCapacity.R:76]: > I assume you checked the output and it contains the right values :-)

yes, I did ^^

File: R/calcCapacity.R (hunk 2 above)
Comment (cchrisgong) [reply R/calcCapacity.R:189]: this was the legacy from when we created a switch for wind offshore, so there are two similar copies of input data for capacity, one with and one without wind offshore.. So both input files are created, and when offshore wind is off in REMIND, it is using a file without wind offshore in it. I thought about making the with windoffshore one default, but then I remember we probably need the switch for DIETER coupling (without the very exogenously defined offshore wind production). This is probably a tangent issue tho - What do you think?

File: R/calcCapacity.R (hunk 1 above)
Comment (cchrisgong) [reply R/calcCapacity.R:76]: > maybe add a comment: "creating empty lines to be filled by hand below"?

Thanks, this works

**Review by LaviniaBaumstark — APPROVED** [review]

File: R/calcCapacity.R (hunk 2 above)
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [reply R/calcCapacity.R:189]: > this was the legacy from when we created a switch for wind offshore, so there are two similar copies of input data for capacity, one with and one without wind offshore

Which is no reason to define identical data twice in two different locations.  Define it once, then drop it into `output` at two points.  Else you will happily change the data for one file at some point in the future, wondering why your change does not affect the other file, which you are looking at.

Comment (cchrisgong) [reply R/calcCapacity.R]: thanks - I cleaned up the redundant scripts - could you check quickly? thx

File: R/calcCapacity.R [hunk 3]
```diff
@@ -72,81 +95,19 @@ calcCapacity <- function(subtype) {
...
-    WEOcap <- rename_dimnames(WEOcap, dim = 3, query = mapping, from = "WEO_techs", to="REMIND_techs")
-    WEOcap <- WEOcap * 1E-03 # converting GW to TW
-
+    
     # merge IRENA, Openmod and WEO capacities data
     output <- new.magpie(cells_and_regions=unique(c(getRegions(IRENAcap),getRegions(Openmodcap),getRegions(WEOcap))),
                          years = unique(c(getYears(IRENAcap),getYears(Openmodcap),getYears(WEOcap))),
-                         names = unique(c(getNames(IRENAcap),getNames(Openmodcap),getNames(WEOcap))),
+                         names = unique(c(getNames(IRENAcap),getNames(Openmodcap),getNames(WEOcap), getNames(CHA.2020.GasData))),
                          fill=0)
```
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [inline R/calcCapacity.R:109]: So you define some magpie object, leave it empty, to get it's "names" (one string) from it?

I'd suggest
```
    CHA.2020.GasData <- as.magpie(
      tribble(
        ~region,   ~year,   ~data,      ~value,
        "CHA",     2010,    "gaschp",   0.022,
        "CHA",     2015,    "gaschp",   0.05,
        "CHA",     2020,    "gaschp",   0.054,
        "CHA",     2010,    "ngcc",     0.001,
        "CHA",     2015,    "ngcc",     0.005,
        "CHA",     2020,    "ngcc",     0.01,
        "CHA",     2010,    "ngt",      0.003,
        "CHA",     2015,    "ngt",      0.016,
        "CHA",     2020,    "ngt",      0.026))
…
    output[getRegions(CHA.2020.GasData),getYears(CHA.2020.GasData),
           getNames(CHA.2020.GasData)] <- CHA.2020.GasData
```

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — APPROVED** [review]
The linter will not be happy with your changes.  But since Lavinia doesn't care …

File: R/calcCapacity.R [hunk 4]
```diff
@@ -96,43 +96,44 @@ calcCapacity <- function(subtype) {
...
     output[getRegions(IRENAcap),getYears(IRENAcap),getNames(IRENAcap)] <- IRENAcap[getRegions(IRENAcap),
                                                                                    getYears(IRENAcap),
                                                                                    getNames(IRENAcap)]
+    
     output[getRegions(Openmodcap),getYears(Openmodcap),getNames(Openmodcap)] <- Openmodcap[getRegions(Openmodcap),
                                                                                            getYears(Openmodcap),
                                                                                            getNames(Openmodcap)]
     output[getRegions(WEOcap),getYears(WEOcap),getNames(WEOcap)] <- WEOcap[getRegions(WEOcap),
                                                                            getYears(WEOcap),
                                                                            getNames(WEOcap)]
```
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [inline R/calcCapacity.R:135]: You changed the order in which data is copied to `output`.   `CHA.2020.GasData` goes first, and then `IRENAcap`, `Openmodcap`, and `WEOcap` all get a chance to overwrite it.
Unless you are sure none of them contain any gas power capacities, that might return something unexpected.

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — CHANGES_REQUESTED** [review]

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — APPROVED** [review]

**Pull request merged**